pennylane_orquestra/gen_workflow.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module contains utilities and auxiliary functions for generating Orquestra
+workflows.
+"""
+# Backend import dictionaries used for generating Orquestra workflows
+
+forest_import = {
+    "name": "qe-forest",
+    "type": "git",
+    "parameters": {"repository": "git@github.com:zapatacomputing/qe-forest.git", "branch": "dev"},
+}
+
+
+qiskit_import = {
+    "name": "qe-qiskit",
+    "type": "git",
+    "parameters": {"repository": "git@github.com:zapatacomputing/qe-qiskit.git", "branch": "dev"},
+}
+
+qulacs_import = {
+    "name": "qe-qulacs",
+    "type": "git",
+    "parameters": {"repository": "git@github.com:zapatacomputing/qe-qulacs.git", "branch": "dev"},
+}
+
+backend_import_db = {
+    "qe-forest": forest_import,
+    "qe-qiskit": qiskit_import,
+    "qe-qulacs": qulacs_import,
+}
+
+
+def step_dictionary(name_suffix):
+    """Creates a new step with a pre-defined name suffixed with the name
+    passed.
+
+    Args:
+        name_suffix (str): The name suffix to use, usually the index of the
+            step. Such an index as suffix can be used for sorting the workflow
+            steps (e.g., for batch execution for the Orquestra device).
+
+    Returns:
+        dict: the dictionary containing information for the step
+    """
+    name = "run-circuit-and-get-expval-" + name_suffix
+    step_dict = {
+        "name": name,
+        "config": {
+            "runtime": {
+                "language": "python3",
+                "imports": [
+                    "pl_orquestra",
+                    "z-quantum-core",
+                    "qe-openfermion",
+                    # Place to insert: step backend component import
```

Comment:
Is this coming in a future PR?

Reply:
Nope, these comments are meant to visually represent where the `gen_expval_workflow` inserts content to create an Orquestra workflow. Just thought this could be helpful when compared with examples from Orquestra, but can remove the comment if it seems confusing.